sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayClientBuilder.java

```diff
@@ -0,0 +1,301 @@
+public final class CommunicationRelayClientBuilder {
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private static final String COMMUNICATION_IDENTITY_PROPERTIES =
+        "azure-communication-networktravesal.properties";
+
+    private final ClientLogger logger = new ClientLogger(CommunicationRelayClientBuilder.class);
+    private String endpoint;
+    private AzureKeyCredential azureKeyCredential;
+    private TokenCredential tokenCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions = new HttpLogOptions();
+    private HttpPipeline pipeline;
+    private RetryPolicy retryPolicy;
+    private Configuration configuration;
+    private ClientOptions clientOptions;
+    private final Map<String, String> properties = CoreUtils.getProperties(COMMUNICATION_IDENTITY_PROPERTIES);
+    private final List<HttpPipelinePolicy> customPolicies = new ArrayList<HttpPipelinePolicy>();
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param pipeline HttpPipeline to use, if a pipeline is not
+     * supplied, the credential and httpClient fields must be set
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = tokenCredential;
+        return this;
+    }
+
+    /**
+     * Sets the {@link AzureKeyCredential} used to authenticate HTTP requests.
+     *
+    * @param keyCredential The {@link AzureKeyCredential} used to authenticate HTTP requests.
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder credential(AzureKeyCredential keyCredential)  {
+        this.azureKeyCredential = keyCredential;
+        return this;
+    }
+
+    /**
+     * Set endpoint and credential to use
+     *
+     * @param connectionString connection string for setting endpoint and initalizing CommunicationClientCredential
+     * @return CommunicationRelayClientBuilder
+     */
+    public CommunicationRelayClientBuilder connectionString(String connectionString) {
+        Objects.requireNonNull(connectionString, "'connectionString' cannot be null.");
```

Comment:
I think this can be an exception since "connectionString" is not global, we can only check it here.

Reply:
We can have another field on the builder called `connectionString`.

Reply:
Removed the check as suggested and actually we don't need to add check for null since ConnectionString adds endpoint and credential, so we have that check already and also there are scenarios where ConnectionString can be null (when credential and endpoint can be set directly)